app/services/data_migrations/add_ethnicity_to_teach_first_trainees.rb

```diff
@@ -0,0 +1,138 @@
+# frozen_string_literal: true
+
+module DataMigrations
+  class AddEthnicityToTeachFirstTrainees
+    def call
+      entries.each do |trainee_id, ethnicity|
+        trainee = teach_first_trainees.where(trainee_id:).first
+
+        next unless trainee
+
+        update_trainee(trainee, ethnicity)
+      end
+    end
+
+  private
+
+    def teach_first_trainees
+      Provider.find_by(code: Provider::TEACH_FIRST_PROVIDER_CODE).trainees
+    end
+
+    def update_trainee(trainee, ethnicity)
+      ethnicity_split = ethnicity.split(":").map(&:strip)
+
+      ethnic_background = ::Hesa::CodeSets::Ethnicities::NAME_MAPPING[ethnicity_split.first]
+      ethnic_group = ::Diversities::BACKGROUNDS.select { |_key, values| values.include?(ethnic_background) }&.keys&.first
+      additional_ethnic_background = ethnicity_split.last
+
+      trainee.update(ethnic_group:, ethnic_background:, additional_ethnic_background:)
+    end
+
+    def entries
+      [
+        ["0034K00000aVaLfQAK", "Any other ethnic background: romanian"],
```

Comment:
Not sure if we should keep this information in an application service. Feels like it should go directly in the data migration since it's a one off data fix

Reply:
I'm really not sure. as this has a bit of logic, not just a straight write, I thought we may need a service to test it? outcome of this ticket https://trello.com/c/M8cCywO0/1404-do-data-migrations-better
<img width="567" alt="Screenshot 2023-12-15 at 10 10 46" src="https://github.com/DFE-Digital/register-trainee-teachers/assets/6339025/88c04c4d-0b94-4793-b092-7ea815c50fa6">

@carlosmartinez any thoughts?

Reply:
Agree it's not really a service, but I do think it should be testable. How about moving this class to be nested in `AddEthnicityToTeachFirstTrainees`?

Reply:
I was actually talking about the data itself eg `["0034K00000aVaLfQAK", "Any other ethnic background: romanian"],` we're hardcoding ids in an application service which is designed to backfill said ids but after that they're pretty irrelevant to the application?

Reply:
gotcha. could pass them as a param from the migration?

Reply:
@carlosmartinez do you mean something like this?

```ruby
class AddEthnicityToTeachFirstTrainees < ActiveRecord::Migration[6.0]
  def up
    Service.call
  end

  def down
    raise ActiveRecord::IrreversibleMigration
  end

  class Service
    include ServicePattern

    def call
      entries.each do |trainee_id, ethnicity|
        trainee = teach_first_trainees.where(trainee_id:).first

        next unless trainee

        update_trainee(trainee, ethnicity)
      end
    end

    private

    def teach_first_trainees
      Provider.find_by(code: Provider::TEACH_FIRST_PROVIDER_CODE).trainees
    end

    def update_trainee(trainee, ethnicity)
      # ... 
    end

    def entries
      # ... 
    end
  end
end
```

Reply:
yes i think i quite like that. what do you reckon?

Reply:
cool, yeah that works, and the specs run fine via `AddEthnicityToTeachFirstTrainees::Service`. looks good to me!